NUT-18: Payment request consolidation (and extra fees for non-preferred mints) (#381)
cashubtc/nuts: 26.md

| 0x02 | amount            | u64       | Amount in base units (corresponds to `a` in JSON)                                                 |
| 0x03 | unit              | u8/string | Currency unit (corresponds to `u` in JSON)                                                        |
| 0x04 | single_use        | u8        | Single-use flag: 0=false, 1=true (corresponds to `s` in JSON)                                     |
| 0x05 | mint              | string    | Mint URL (repeatable for multiple mints, corresponds to `m` in JSON)                              |
| 0x06 | description       | string    | Human-readable description (corresponds to `d` in JSON)                                           |
| 0x07 | transport         | sub-TLV   | Transport configuration (repeatable, corresponds to `t` in JSON)                                  |
| 0x08 | nut10             | sub-TLV   | NUT-10 spending conditions (corresponds to `nut10` in JSON)                                       |
| 0x09 | mint_preferred    | u8        | Mint list strictness flag: 0=false, 1=true; defaults to 0 if absent (corresponds to `mp` in JSON) |

> | 0x0a | fee_reserve       | u64       | Additional fee reserve for payments from non-preferred mints (corresponds to `fr` in JSON)        |

Reviewer: ```suggestion
| 0x0a | method_fee       | u64       | Additional fee for each payment method (corresponds to `fr` in JSON)        |
```
maybe something like this

Author: I think you are mixing two semantics here. `fr` is the additional "fee required" if using a non-preferred mint.

A per-method fee would be a different thing to consider. For example, if there are three supported methods (bolt11, bolt12, onchain) how should we represent a fee for each method?

Edit: Idea here: https://github.com/cashubtc/nuts/pull/381#discussion_r3501713194

> | 0x0b | supported_methods | string    | Supported payment method (repeatable for multiple methods, corresponds to `sm` in JSON)           |

Reviewer: ```suggestion
| 0x0b | payment_method | string    | Supported payment method (repeatable for multiple methods, corresponds to `sm` in JSON)           |
```

Author: The `supported_methods` label matches the `sm` key. It should be singluar (as repeatable in nut26), so I changed the label to `supported_method`. No changes to implementations required.
